tests/baselines/reference/templateLiteralTypes2.types

```diff
@@ -402,13 +402,13 @@ const interpolatedStyle = { rotate: 12 };
 
 function C2(transform: "-moz-initial" | (string & {})) { return 12; }
 >C2 : (transform: "-moz-initial" | (string & {})) => number
->transform : (string & {}) | "-moz-initial"
+>transform : string
 >12 : 12
 
 C2(`rotate(${interpolatedStyle.rotate}dig)`);
 >C2(`rotate(${interpolatedStyle.rotate}dig)`) : number
->C2 : (transform: (string & {}) | "-moz-initial") => number
->`rotate(${interpolatedStyle.rotate}dig)` : `rotate(${number}dig)`

```

Comment:
Hm, so, this actually could be hugely breaking in the community - `string & {}` was an intentional workaround here - when people wanted to say "this takes a string, but here's a union of well-known members so you get good string completions for well-known options", we told them "oh, write `string & {}` instead of string and then it won't subtype reduce with the literals", and so this pattern is insanely common in the wild. Obviously there won't be any assignability changes from actually reducing it, but there could be conditional type execution differences (in some edge cases) and there definitely will be completion fidelity differences (which, unfortunately, won;t be picked up by most of our extended suites since it's a services feature). AFAIK, this pattern's in use all over the css types and react types.

Reply:
Lovely. I guess every cute trick comes back to bite us eventually. Luckily it's pretty easy to implement exceptions that preserve an explicitly written `string & {}` intersections (and similar ones for `number` and `bigint`). I'll look at doing that.

Reply:
May need to expand that further - csstype exports a conditional type like
```
export type PropertyValue<TValue> = TValue extends Array<infer AValue>
  ? Array<AValue extends infer TUnpacked & {} ? TUnpacked : AValue>
  : TValue extends infer TUnpacked & {}
  ? TUnpacked
  : TValue;
```
whose explicit purpose seems to be unwrapping `... & {}` types. In this case, I don't think the result changes, but I'd be worried for other conditionals in the wild.

Perhaps more importantly, in https://github.com/microsoft/TypeScript/issues/29729 the first workaround is
```ts
type LiteralUnion<T extends U, U = string> = T | (U & {});
```
-a quick code search on DT shows at least 8 projects using either that generic pattern verbatim or similar workarounds.

Reply:
Oh, it's in `type-fest`, [too](https://github.com/sindresorhus/type-fest/blob/main/source/literal-union.d.ts#L32), which is used by sindresoruh's projects and beyond - (`Record<never, never>` is `{}` with more characters because eslint bans `{}` for some reason and it's a cheeky workaround)

Reply:
Only two of the `LiteralUnion<...>` occurrences on DT use `T & {}`. The others are `T & Record<never, never>` or `T & {  _?: never | undefined }` which continue to work as before. Seems manageable to just fix those two.

Reply:
I'm not a huge fan of `Record<never, never>` and `{}` behaving differently just because one happens to lack the `Anonymous` object flag, specifically since `Record<never, never>` is used as an alias to `{}` in the wild thanks to `eslint`. For the same reason we see `T & Record<never, never>` in these generic types, we're likely to see `string & Record<never, never>` directly in the wild (which is to say: to satisfy the linter).

Reply:
See #49220.